File: Greedy/greedy_1.go
package main

import (
	"fmt"
	"sort"
)

func solution(n int, m int, k int, nums []int) int {
	var result int = 0
	sort.Slice(nums, func(i, j int) bool {
		return nums[i] > nums[j]
	})

	result += nums[0] * int(m/k) * k
	result += nums[1] * (m % k)

	return result
}

func main() {
	var n, m, k int
	fmt.Scan(&n, &m, &k)

	nums := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&nums[i])
	}

	var s int = solution(n, m, k, nums)
	fmt.Println(s)
}

File: Greedy/greedy_1.py
# 큰 수의 법칙
## 배열의 크기 N, 숫자가 더해지는 횟수 M, 연속 횟수 제한 K가 주어질 때 큰 수의 법칙에 따른 결과를 출력하시오
### 첫째 줄에 N, M, K의 자연수가 주어지며, 각 자연수는 공백으로 구분한다. (K는 M보다 작거나 같다)
### 둘째 줄에 N개의 자연수가 주어지며, 각 자연수는 공백으로 구분한다.
#### 큰 수의 법칙에 따라 더해진 값 출력

def solution(n, m, k, nums):
    result = 0
    nums.sort(reverse = True)

    result += nums[0] * (m//k) * k
    result += nums[1] * (m%k)
    
    return result

n, m, k = map(int, input().split())
nums = list(map(int, input().split()))

s = solution(n, m, k, nums)
print(s)